LNLS site-specific config

laispc opened the pull request:
Hi,

This is a PR to share LNLS site-specific code (classes, config). Hopefully it will not affect much of general code (if it happens we can treat in separated PR maybe). This is the code we have been using for a while now (more recent code will come next).

There is still some fixes to do (clean-up, squashes, code style), so I am putting this as a draft PR for now. But feel free to make comments and suggestions anytime, as the main purpose here is to share and receive critique on the work done. Thanks in advance! ;)

IvarsKarpics commented on HardwareObjects/GenericDiffractometer.py:
```diff
@@ -890,7 +890,7 @@ def centring_done(self, centring_procedure):
                     "Centring finished. Moving motoros to position %s" % str(motor_pos)
                 )
-                self.move_to_motors_positions(motor_pos, wait=True)
+                self.move_to_motors_positions({}, wait=True)
```
Hmm, why this was removed?

laispc replied:
@IvarsKarpics it was done just for a test, it was intended to be reverted and does not apply anymore, sorry.
I will drop this change, thanks!

beteva commented on HardwareObjects/LNLS/LNLSInOut.py:
```diff
@@ -0,0 +1,124 @@
+import logging
```
@laispc, this code is just fine. Unfortunately with the new HardwareRepository structure we decided to drop the inheritance from Device and Equipment and only use HardwareObject. Moreover, for InOut, there is AbstractNState. If you want to use the new HardwareRepository structure, you may want to change this to LNLSNState, which inherits from AbstractNState. You could see ExporterNState as an example

laispc replied:
Thanks for the info! I will try to refactor this part.

laispc replied:
@beteva LNLSInOut refactored, now it is called EPICSNState, following the suggested structure.

beteva commented on HardwareObjects/LNLS/LNLSMotor.py:
```diff
@@ -0,0 +1,89 @@
+        """Override super class method."""
+        self.__velocity = self.set_channel_value(self.MOTOR_VELO, value)
+
+    def validate_value(self, value):
+        """Override super class method."""
+        try:
+            value = float(value)
+        except Exception as e:
+            pass
+        if value is None:
+            return True
+        if math.isnan(value) or math.isinf(value):
+            return False
+        limits = self._nominal_limits
+        if None in limits:
```
The same code is in AbstractMotor, except for the casting of value to float. Do you really need it?

laispc replied:
I remember duplicating this part due to a inheritance bug on this class, but I will check again if it's still necessary.

laispc replied:
I just removed this piece of code from LNLSMotor. It is not needed anymore. :)

Btw, LNLSMotor was renamed to EPICSMotor.

beteva commented on HardwareObjects/LNLS/LNLSTransmission.py:
```diff
@@ -0,0 +1,89 @@
+    def init(self):
+        """Override method."""
+        AbstractTransmission.init(self)
+        self.energy = self.getObjectByRole("energy")
```
you could use:
from HardwareRepository import HardwareRepository as HWR
and then self.energy could be replaced by HWR.beamline.energy
Thus the energy HO will only be defined in beamline_config.yml for any place you want to use the energy

laispc replied:
Great! It could make the code/config more simple indeed.

laispc replied:
@beteva I just implemented you suggestion in LNLSTransmission and LNLSCollect. It worked fine, thanks!

laispc commented:
@beteva thanks for the review so far. I am working on them and testing.

@IvarsKarpics thank you too, I already applied the changes you recommended.

IvarsKarpics commented on configuration/lnls_manaca/detector_mockup.xml:
```diff
@@ -0,0 +1,31 @@
+<equipment class = "LNLS.LNLSPilatusDet">
```
maybe rename the config file to detector.xml?

laispc replied:
Fixed.

IvarsKarpics commented on configuration/lnls_manaca/ldapconnection.esrf.xml:
```diff
@@ -0,0 +1,4 @@
+<procedure class="LdapLogin">
```
?

marcus-oscarsson replied:
yes the .esrf. name is a bit funny :)

laispc replied:
Lol, I have borrowed the XML names from the mock config. I am renaming them. :) Thank you!

laispc replied:
All files renamed.

IvarsKarpics reviewed (changes requested):
Apart from few comments and swp file in the commit, the PR looks fine.

marcus-oscarsson commented:
Thanks for this really nice work @laispc :), @IvarsKarpics had a few final remarks that I think is worth having a look at.  I also think the PR is basically ready to be merged. Let us know when you are done with the last fixes :)

laispc commented on Command/Epics.py:
```diff
@@ -1,29 +1,32 @@
+import saferef
+#from . import Poller
+#from Poller import *
+import Poller
```
Fixed.

laispc commented on Command/Epics.py:
```diff
@@ -1,29 +1,32 @@
+#from Poller import *
+import Poller
+#from .CommandContainer import CommandObject, ChannelObject
+from CommandContainer import CommandObject, ChannelObject
```
Fixed.

laispc commented:
@beteva @IvarsKarpics @marcus-oscarsson , I just finished the suggested changes.
Feel free to take a look and make more suggestions if you want. Thank you! :)

IvarsKarpics commented:
Hi @laispc 

There is still one swap file in the commit. After removing it the PR is ready to be merged.

laispc commented:
> There is still one swap file in the commit. After removing it the PR is ready to be merged.

Hi @IvarsKarpics , shame on me, I removed the.swp from the PR, but I think it did not work in the end, sorry. 
I just fixed it and I think it is okay now. Thanks!

IvarsKarpics commented:
Hi @laispc 

I see, now the swap file is gone. However I see that the travis is failing. Looks like there are few typos. Can you fix them so the travis is not failing?

laispc commented:
@IvarsKarpics yes, there was a typo in Command/Epics.py. Fixed!

Also, LNLS/test folder was being called by Travis by accident, as it has 'test' in the name. As it wasn't my intention to make LNLS/test be called by pytest yet, I am dropping LNLS/test from this PR because it is not critical. Travis is happy again. :)

However, this error it helped me to actually see the typo and that Command/Epics.py is not tested by Travis. As a future work we could try to add a test for Command/Epics.py. Does it make sense for HardwareRepository?

IvarsKarpics commented:
Hi @laispc 

Thanks for fixing the bug. Initially we where running pylint -E against all repository files, but at some point there was so many errors that we could not solve them easy. I think when we are ready with renaming of the repository we can check again why pylint is failing.
Also we can configure pytest so it is not running any other tests as those defined in the tests directory.

I am merging the PR.
